Backend setup for updating events in adminPage

CraftyChimera opened the pull request.

CraftyChimera committed:
[feat] init: admin page backend

CraftyChimera committed:
feat(adminpage): Improved Code structure and Added tests

CraftyChimera committed:
[feat] Added Auth backend

kelpikz commented on server/controllers/event.py:
```diff
@@ -63,3 +65,43 @@ def get_events(
+    """
+    for events in events_list:
+        event_data = (
+            database.query(Event).filter_by(name=events.name).first()
```
how will you update event names?

kelpikz commented on server/controllers/event.py:
```diff
@@ -63,3 +65,43 @@ def get_events(
+        if not event_data:
+            raise GenericError("Event Does not exist")
+
+        point_list = (
+            database.query(Point).filter_by(event_id=event_data.id).all()
+        )
```
can you add validation to check if the length of ```point_list``` and ```events.points``` is 3

kelpikz commented on server/controllers/event.py:
```diff
@@ -63,3 +65,43 @@ def get_events(
+                raise GenericError("Invalid Department")
+            item.department_id = department_id
+
+        event_data.description = events.description
+        event_data.rules = events.rules
+        event_data.form_link = events.form_link
+        event_data.event_link = events.event_link
+        event_data.image_link = events.image_link
+        event_data.start_time = events.start_time
+        event_data.end_time = events.end_time
+        event_data.is_reg_completed = events.is_reg_completed
+        event_data.is_event_completed = events.is_event_completed
```
can we have point updation and event updation as 2 separate routes ?

kelpikz commented on server/routers/admin.py:
```diff
@@ -58,17 +70,40 @@ async def download_responses(
+        jwt_res = sign_jwt_auth(
+            roll=response.roll_number, password=response.password
+        )
+        print(jwt_res)
```
remove unnecessary print statement

CraftyChimera committed:
[refactor] Fixed some bugs and rearranged code:
- Split route for Update into routes for Update_Events and Update_points
- Added validation checks in places

CraftyChimera committed:
[refactor] linting fixes
